R/calculate_company_alignment_metric.R

```diff
@@ -354,54 +358,81 @@ calculate_company_aggregate_alignment_sda <- function(data,
     scenario = scenario
   )
 
+  # params
   start_year <- min(data$year, na.rm = TRUE)
+  # standard forward looking PACTA time frame
+  time_frame <- 5
   target_scenario <- paste0("target_", scenario)
+  # since sda sectors are not split into technologies, the level is always: "net"
+  level <- "net"
 
+  # prep and wrangle
   data <- data %>%
-    dplyr::filter(.data$scenario_source == .env$scenario_source)
+    prep_and_wrangle_aggregate_alignment_sda(
+      scenario_source = scenario_source,
+      target_scenario = target_scenario,
+      start_year = start_year,
+      time_frame = time_frame
+    )
+
+  # add activity units to data
+  activity_units_sector <- activity_units %>%
+    dplyr::distinct(.data$sector, .data$activity_unit)
 
   data <- data %>%
-    group_by(
-      .data$group_id, .data$name_abcd, .data$emission_factor_metric, .data$year, .data$region,
-      .data$scenario_source
-    ) %>%
+    dplyr::inner_join(activity_units_sector, by = "sector")
+
+  # add tech share by direction
+  data <- data %>%
+    add_technology_share_by_direction(level = level)
+
+  # calculate alignment metric
+  data <- data %>%
+    add_net_absolute_scenario_value(target_scenario = target_scenario) %>%
+    add_total_deviation_sda() %>%
+    calculate_company_alignment_metric(scenario = scenario)  %>%
+    dplyr::arrange(
+      .data$group_id,
+      .data$sector,
+      .data$name_abcd,
+      .data$region,
+      .data$year
+    )
+
+  return(data)
+}
+
+prep_and_wrangle_aggregate_alignment_sda <- function(data,
+                                                     scenario_source,
+                                                     target_scenario,
+                                                     start_year,
+                                                     time_frame) {
+  data <- data %>%
+    dplyr::filter(.data$scenario_source == .env$scenario_source) %>%
     dplyr::filter(.data$name_abcd != "market") %>%
-    dplyr::filter(.data$emission_factor_metric %in% c("projected", paste0("target_", .env$scenario))) %>%
-    dplyr::filter(dplyr::between(.data$year, left = .env$start_year, right = .env$start_year + 5)) %>%
+    dplyr::filter(.data$emission_factor_metric %in% c("projected", .env$target_scenario)) %>%
+    dplyr::filter(
+      dplyr::between(
+        .data$year,
+        left = .env$start_year,
+        right = .env$start_year + .env$time_frame)
+    ) %>%
     tidyr::pivot_wider(
       names_from = "emission_factor_metric",
       values_from = "emission_factor_value"
-    ) %>%
-    dplyr::ungroup()
```

Comment:
looks like you lost this `ungroup()` in the new chain, which may or may not be important

Reply:
this is intentional, the corresponding `group_by()` is also being removed. It seems to be a remainder from a previous version of the function and I tested that removing it has no effect.